src/Document.ts

```diff
@@ -199,12 +199,18 @@ class Document {
 						setIn(this, keyPath, value);
 
 						const errors = await schemaType.validate(value, this);
-						if (errors.length > 0) {
+						if (errors instanceof Map) {
+							errors.forEach((error, key) => {
```

Comment:
`error` is a bit confusing of a variable name here since its an array. I would suggest using a name that implies that.